src/main/java/duke/Duke.java

```diff
@@ -384,85 +406,78 @@ public void remove(Task t) {
     public class Parser {
 
         /**
-         * Parses a given input and performs actions as necessary based on the input.
+         * Parses a given input and returns Strings as necessary based on the input.
          * The main driver of the Duke class.
          *
          * @param leest The current TaskList.
          * @param input Input that is being read in.
          */
-        public void parse(TaskList leest, String input) {
+        public String parse(TaskList leest, String input) {
```

Comment:
Haha cute name, but maybe you could consider changing it to something more understandable like taskList?